ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutor.java

```diff
@@ -355,12 +355,12 @@ private void terminateQuery(final PreparedStatement<TerminateQuery> terminateQue
     final Optional<QueryId> queryId = terminateQuery.getStatement().getQueryId();
 
     if (!queryId.isPresent()) {
-      ksqlEngine.getPersistentQueries().forEach(PersistentQueryMetadata::close);
+      ksqlEngine.getPersistentQueries().forEach(q -> q.close(true));
       return;
     }
-
+    
     final Optional<PersistentQueryMetadata> query = ksqlEngine.getPersistentQuery(queryId.get());
-    query.ifPresent(PersistentQueryMetadata::close);
+    query.ifPresent(query -> query.close(true));
```

Comment:
This line causes compilation failure.

```suggestion
    query.ifPresent(q -> q.close(true));
```